### PR TITLE
Fix returning/yielding chat message from callback

### PR DESCRIPTION
**Pull request by ahuang11**
Closes https://github.com/holoviz/panel/issues/5708

### CODE REVIEW

File: panel/chat/feed.py [hunk 1]
```diff
@@ -335,12 +335,17 @@ def _upsert_message(
         if isinstance(value, dict):
             user = value.get("user", user)
             avatar = value.get("avatar")
+
         if message is not None:
```
Comment (MarcSkovMadsen) [inline panel/chat/feed.py:339]: Could there be a situation where `message is not None and isinstance(value, ChatMessage)`? And how should that be handled?

Comment (ahuang11) [reply panel/chat/feed.py:339]: ChatMessage overrides it I think in

        elif isinstance(value, ChatMessage):
            # ChatMessage is not created yet, but a ChatMessage is passed; use it
            self._replace_placeholder(value)
            return value

**Comment by MarcSkovMadsen** [comment]
I've tested that `async` and `return` works.

But `async` and `yield` does not seem to work.

```python
import panel as pn
import pandas as pd

agent_user="Pandas Agent"
agent_avatar="🐼"

pn.extension("perspective")

def _create_message(message):
    return pn.chat.ChatMessage(message, user="Pandas Agent", avatar="🐼")

async def callback(contents, user, instance):
    if isinstance(contents, pd.DataFrame):
        instance.active=1
        print("I was here")
        yield _create_message("You uploaded a dataframe. Ask me anything about it")
        

    yield _create_message("That is a great question. I don't know.")

chat_interface = pn.chat.ChatInterface(
    widgets=[pn.widgets.FileInput(name="Upload"), pn.widgets.TextInput(name="Message")],
    renderers=pn.pane.Perspective,
    callback=callback,
    callback_exception="verbose",
)
    
chat_interface.send("I'm your LangChain Pandas Agent!", user="Pandas Agent", avatar="🐼", respond=False)
chat_interface.servable()
```

image

File: panel/chat/feed.py (hunk 1 above)
Comment (MarcSkovMadsen) [reply panel/chat/feed.py:339]: I can't see that. Where do you see that?

image

Comment (ahuang11) [reply panel/chat/feed.py:339]: :) Ok, perhaps not ChatMessage, but dict works as expected.

**Comment by ahuang11** [comment]
Should be fixed now.

File: panel/chat/feed.py (hunk 1 above)
Comment (MarcSkovMadsen) [reply panel/chat/feed.py:339]: Should you add something like?

```python
...
elif isinstance(value, ChatMessage):
    user = value.user
    avatar = value.avatar
    value = value.value
```

**Comment by MarcSkovMadsen** [comment]
> Should be fixed now.

I can confirm its fixed.Thanks.

File: panel/chat/feed.py (hunk 1 above)
Comment (ahuang11) [reply panel/chat/feed.py:339]: No, we don't need to extract the values if it's a ChatMessage and message is None.

It does look neater though.

**Review by philippjfr — APPROVED** [review]

**Pull request merged**